### PR TITLE
[HTML5] Fix build (with module_webxr_enabled=no). Re-add it to CI.

### PR DESCRIPTION
**Pull request by Faless**
In this PR:

- Fix build issues for the javascript platform after various refactoring.
- Re-add javascript to CI, only thread-enabled build, and with WebXR disabled (XR API changed, will need fixing).

### CODE REVIEW

File: .github/workflows/javascript_builds.yml [hunk 1]
```diff
@@ -4,9 +4,9 @@ on: [push, pull_request]
 # Global Settings
 env:
   GODOT_BASE_BRANCH: master
-  SCONSFLAGS: platform=javascript verbose=yes warnings=extra werror=yes debug_symbols=no --jobs=2
+  SCONSFLAGS: platform=javascript verbose=yes warnings=extra werror=yes debug_symbols=no module_webxr_enabled=no threads_enabled=yes --jobs=2
   SCONS_CACHE_LIMIT: 4096
-  EM_VERSION: 1.39.20
+  EM_VERSION: 2.0.15
```
Comment (akien-mga) [inline .github/workflows/javascript_builds.yml:9]: I just updated my containers to 2.0.25 (will be used for 3.4 and 4.0, unless we need to rollback) so we could use it here too.

**Review by akien-mga — APPROVED** [review]
Seems good. Since thread support is now mandatory, it should probably be made the default option? (Or outright remove the `threads_enabled` setting.)

I guess GDNative is broken but that's expected for now (and we'll need it to eventually work with thread support to make a comeback).

**Comment by Faless** [comment]
> Seems good. Since thread support is now mandatory, it should probably be made the default option? (Or outright remove the `threads_enabled` setting.)

Done, and updated to emcc `2.0.25`. Had to also remove `werror=yes` due to warnings about `ordered comparison of function pointers`. Should I instead disable it somewhere in `scons` via `-Wno-ordered-compare-function-pointers`?

**Comment by akien-mga** [comment]
> Should I instead disable it somewhere in `scons` via `-Wno-ordered-compare-function-pointers`?

Well the best would be to fix the warning, it seems to point at a valid issue. But until then we can disable werror for javascript yeah.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**